packages/vitest/src/runtime/collect.ts

```diff
@@ -64,7 +64,8 @@ export async function collectTests(paths: string[], config: ResolvedConfig) {
 
     calculateHash(file)
 
-    interpretTaskModes(file, config.testNamePattern)
+    const hasOnlyTasks = someTasksAreOnly(file)
+    interpretTaskModes(file, config.testNamePattern, hasOnlyTasks, false, config.allowOnly)
```

Comment:
Do we need to pass `allowOnly` around? 
Looks like this be simplified to:
```js
const hasOnlyTasks = someTasksAreOnly(file)
if (hasOnlyTasks && !config.allowOnly) {
   file.result = {
      state: 'fail',
      error: processError(new Error('only is not allowed')),
   }
}
else {
   interpretTaskModes(...
}
```

Reply:
That depends on the behavior we want here.

With the current implementation, the locations of `only` and thus the failing tasks are explicitly listed.
If we just fail the entire file there's less context about the source.

Reply:
Makes sense